troi/__init__.py

```diff
@@ -97,7 +94,7 @@ def outputs():
         return None
 
     @abstractmethod
-    def read(self, source_data_list, debug):
+    def read(self, source_data_list):
```

Comment:
I still missed some? Sheesh!

Reply:
I think I got them all this time. It still exists in the constructor of Patch, which I believe is the only place that it needs to be, right?

Reply:
Correct.